subprojects/language-scala/src/main/java/org/gradle/language/scala/tasks/AbstractScalaCompile.java

```diff
@@ -89,7 +91,12 @@ public CompileOptions getOptions() {
     protected void compile() {
         ScalaJavaJointCompileSpec spec = createSpec();
         configureIncrementalCompilation(spec);
-        getCompiler(spec).execute(spec);
+        Compiler<ScalaJavaJointCompileSpec> compiler = getCompiler(spec);
+        File analysisFile = getScalaCompileOptions().getIncrementalOptions().getAnalysisFile();
+        if(analysisFile==null){
```

Comment:
I guess you need to check if the file exists - not if the location of the analysis file is configured.

The location of the analysis file is configured by default.